### DISCUSSION TITLE
cyrillic text and fonts at ruffle

### DISCUSSION THREAD

**Discussion by saslo4r** [ruffle-rs/ruffle#6288] [Ideas]
hello, can you tell me when you will add support for cyrillic text and some fonts?
example, Nikopol district (I am from Ukraine) in place of the gray field should be text:
http://eye.swfchan.com/flash.asp?id=243786&n=nikopol.swf

**Comment by adrian17** [comment]
Actually, I'm pretty sure we just happened to extend the built-in font with cyrylic characters 5 days ago :)
Try downloading this SWF and opening it in https://ruffle.rs/demo/ to see if it works.

**Reply by saslo4r** [reply]
no that's not works :\

**Comment by torokati44** [comment]
That extension did not cover Cyrillic, only Latin-based characters, sorry. 😐

**Reply by adrian17** [reply]
Oh, you're right, I tested the movie but didn't realize there's also on-hover text - and indeed it doesn't show up :( 

How much would adding it "cost"?

**Reply by torokati44** [reply]
The font file would grow to about 96 KiB, so an additional ~25 KiB on top of the current ~72 KiB. Up from the original ~12KiB.
If we wanted to zip/deflate/cmpress it, and unzip on load, it would be about 59 KiB (depending on compression of course) instead of 96 KiB.
EDIT: Although I imagine in at least the self-hosted cases, `Accept-Encoding: br, gzip, compress` would make it not matter much?

**Reply by adrian17** [reply]
Size doesn't sound like a big deal, I'm more worried about having to parse this slowly-increasing font tag for every single movie, not just ones that need these glyphs :/ 
(we can continue the technical discussion on discord if you prefer)

**Reply by saslo4r** [reply]
that's too bad😕

**Reply by saslo4r** [reply]
I'm not on Discord
it remains to download either the old flash player or with the help of patchers to change player code🗿👍

**Reply by torokati44** [reply]
I think he meant that mostly for me. 😅

**Reply by torokati44** [reply]
Thanks for your measurements on Discord! FWIW I vote for further extending the embedded font with these characters.
